src/trip_kinematics/URDFParser.py

```diff
@@ -0,0 +1,301 @@
+        # (Note that the movement of the joint is applied between sta and unsta)
+        unalign_transformation = np.linalg.inv(align_transformation)
+        unalign_quat = ScipyRotation.from_matrix(unalign_transformation).as_quat()
+        unsta = [name + '_unsta',
+                 {'qw': unalign_quat[3],
+                  'qx': unalign_quat[0],
+                  'qy': unalign_quat[1],
+                  'qz': unalign_quat[2]},
+                 [],
+                 ]
+
+        joint_transformations.append(sta)
+        # unsta will be appended after the joint movement transformation
+
+    # Transformation 4 (mov): represents the movement of the joint. This is the only transformation
+    # with state variables.
+    if type_ in ['continuous', 'revolute']:
+        mov = [name + '_mov', {'rz': 0}, ['rz']]
+    elif type_ == 'prismatic':
+        mov = [name + '_mov', {'tz': 0}, ['tz']]
+    # Commenting out floating and planar joints for now because we cannot test them by comparing
+    # against kinpy (since it does not support them). In theory this code should work, though.
+    # elif type_ == 'floating':
+    #     mov = [name + '_mov',
+    #            {'tx': 0, 'ty': 0, 'tz': 0, 'rx': 0, 'ry': 0, 'rz': 0},
+    #            ['tx', 'ty', 'tz', 'rx', 'ry', 'rz']]
+    # elif type_ == 'planar':
+    #     mov = [name + '_mov', {'tx': 0, 'ty': 0}, ['tx', 'ty']]
+    elif type_ == 'fixed':
+        mov = [name + '_mov', {}, []]
+    else:
+        raise ValueError(f"Unsupported joint type {type_}")
+
+    joint_transformations.append(mov)
+
+    if axis is not None and not np.array_equal(axis, np.aray([0, 0, 1])):
+        joint_transformations.append(unsta)
+
+    return joint_transformations
+
+
+def _create_transformations_from_tree(joint: str,
+                                      joint_tree_dict: Dict[str, Dict],
+                                      joint_name_to_transformations: Dict[str, List],
+                                      parent: Transformation) -> List[Transformation]:
+    """Recursively builds a tree of py:class`Transformation` objects, starting from the root and
+    traversing the tree towards the children.
+
+    Args:
+        joint (str): Name of the joint.
+        joint_tree_dict (Dict[str, Dict]): Represents the relationships between all joints.
+        joint_name_to_transformations (Dict[str, List]): Contains the parameters for the
+                                                         py:class`Transformation` objects.
+        parent (Transformation, optional): The parent of current joint. Should be None for the root,
+                                           is set recursively for its children.
+
+    Returns:
+        List[Transformation]: List of py:class`Transformation` objects for input node and all its
+                              descendants.
+    """
+    transformations_list = []
+
+    for trans in joint_name_to_transformations[joint]:

```

Comment:
The Rest of the library tends to use `for transformation in` to loop over transformations.
Please change this to be consistent with the rest of the library